### DISCUSSION TITLE
Help with crazy barometer

### DISCUSSION THREAD

**Discussion by checc0** [iNavFlight/inav#7188] [Q&A]
Hi. I'm using a Matek MQ8 on a GEP-20A-F4 AIO connected on a serial port in order to have (baro+gps+mag) in just one port.
I'm facing a strange issue where sometime the baro doesn't work and I can see on the sensor panel that the value "baro" is always off:

![image](https://user-images.githubusercontent.com/86500321/123492434-c3a79b80-d619-11eb-865a-30fbd7d2c3e8.png)

However sometimes the baro works and what I've noticed by looking at the debug trace that: 

When the it works I've got values around (1700-2000) in the Barometer calibration complete debug line.. viceversa when doesn't works I get  value 4433000 !

I've checked wires 1000 times.. tried with inav 2.6/3.0/31... same result !

What is the meaning of the value in the debug message "Barometer calibration complete (value)" ?
Why I'm getting those high values ?!?
Any idea ?

Thanks in advance !

**Comment by DzikuVx** [comment]
This will be fixed in 3.0.1. Until then, please insert SD card into SD card slot

**Reply by checc0** [reply]
Thanks ! But my board doesn't have any sd card !

**Comment by DzikuVx** [comment]
Then you have to wait until mid Juli when official fix will be released

sob., 26 cze 2021, 13:42 użytkownik checc0 ***@***.***>
napisał:

**Reply by checc0** [reply]
There is a fix for this in an upcoming release ? It's an issue already tracked ? If yes, could you kindly provide more info ?

**Reply by avsaase** [reply]
Yes this is fixed for 3.0.1

**Reply by checc0** [reply]
are you talking about this ? https://github.com/iNavFlight/inav/pull/7182/commits/47f2468f0ec95270f5648c57f2bf9b5fc1ee4529 

If yes, I applied the code to 3.0 but nothing changes :-(

**Reply by avsaase** [reply]
Please explain the exact step you took in https://github.com/iNavFlight/inav/issues/7163.

**Reply by checc0** [reply]
What I've discovered/tried so far:

1) Applying the code fix for the zero calibration issue described here (without success): https://github.com/iNavFlight/inav/pull/7182/commits/47f2468f0ec95270f5648c57f2bf9b5fc1ee4529

2) Apply the following workaround (or I should say porkaround :-D )  in calibration.c:

int32_t baroCalculateAltitude(void)
{
    if (!baroIsCalibrationComplete()) {
        zeroCalibrationAddValueS(&zeroCalibration, baro.baroPressure);
        //LOG_D(BARO, "Barometer pressure(%ld), alt(%ld), temp(%ld)", baro.baroPressure, baro.BaroAlt, baro.baroTemperature);
        if (zeroCalibrationIsCompleteS(&zeroCalibration)) {
            zeroCalibrationGetZeroS(&zeroCalibration, &baroGroundPressure);
            baroGroundAltitude = pressureToAltitude(baroGroundPressure);
            LOG_D(BARO, "Barometer calibration complete (%d)", (int)lrintf(baroGroundAltitude)); 

        }

        // FDE Porkaround  
        if (baroGroundAltitude > 90000) {
            fcReboot(false);
        }

        baro.BaroAlt = 0;
    }
    else {
...

Basically I'm forcing the fc to reboot if the zero calibration failed... i'm still investigating if it's related to an hw or sw issue.
